### PR TITLE
Designed new API and simple example

### PR DESCRIPTION
**Pull request by JohannesMeierSE**
* joint work of @insafuhrmann and @JohannesMeierSE
* Fulfills #24 and #39
* All existing examples including OX and LOX are reworked to use the new factory API
* The factory API is provided for these types: primitive, top, bottom, function, class
* These types are not (yet) supported by the new factory API
    * multiplicity types: I am not sure, whether we really need this type, we should not advertise it for v0.1, but rework/redesign/improve it for upcoming versions
    * fixed parameter type: I am unsure at the moment, how to integrate it into the API, since each kind instance is specific for the current set of operators, e.g. there is a "Map kind" and another "List kind", which "both are FixedParameterKinds" (https://github.com/TypeFox/typir/pull/43/files#diff-06d7c3de2afd260ac79d91a52f56faacc286892f4786c7e7bfbd670494d97429R28)

### CODE REVIEW

File: packages/typir/src/kinds/function/function-kind.ts [hunk 1]
```diff
@@ -96,6 +96,12 @@ export type InferFunctionCall<T = unknown> = {
  */
 
 
+export interface FunctionPredefinedService {
```
Comment (insafuhrmann) [inline packages/typir/src/kinds/function/function-kind.ts:99]: Should this stay "predifined"? The others are named factory

**Review by insafuhrmann — APPROVED** [review]
Thanks for the great joint work session @JohannesMeierSE and for the additions and fixes afterwards. The Functions escaped the renaming from predefined to factory, otherwise everything seems fine to me.

File: packages/typir/src/kinds/function/function-kind.ts (hunk 1 above)
Comment (JohannesMeierSE) [reply packages/typir/src/kinds/function/function-kind.ts]: Yes, I missed that, thanks!

**Comment by JohannesMeierSE** [comment]
Thanks @insafuhrmann for the great joint working session!

**Pull request merged**